ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/SandboxOnXRunner.scala

```diff
@@ -285,16 +287,24 @@ object SandboxOnXRunner {
   }
 
   def buildRateLimitingInterceptor(
```

Comment:
I would do:
```
  def buildRateLimitingInterceptor(
      metrics: Metrics,
      indexDbExecutor: QueueAwareExecutionContextExecutorService,
      apiServicesExecutor: QueueAwareExecutionContextExecutorService,
  )(config: RateLimitingConfig): RateLimitingInterceptor = {

    val indexDbCheck = ThreadpoolCheck(
      "Index DB Threadpool",
      indexDbExecutor,
      config.maxApiServicesIndexDbQueueSize,
    )

    val apiServicesCheck = ThreadpoolCheck(
      "Api Services Threadpool",
      apiServicesExecutor,
      config.maxApiServicesQueueSize,
    )

    RateLimitingInterceptor(metrics, config, List(indexDbCheck, apiServicesCheck))

  }
```

Reply:
We will also have to fix this up on the Canton side, but this is fine as they create their own threadpools so it makes sense for them to create the threadpool checks (as is currently the case).

Reply:
Ah alright, I'm much in favour of this approach. Chose not to go down that path as I assumed we want to enforce the indexer thread pool check.